opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -484,6 +484,75 @@ namespace Opm {
             return istlSolver().iterations();
         }
 
+
+        void makeOverlapRowsInvalid(Mat& ebosJacIgnoreOverlap) const
+        {
+	  //local id's of cells
+	  auto lid = grid_.localIdSet();
+	  
+	  //value to set on diagonal
+	  double diag_val = 1.0e100;
+	  
+	  const auto& gridView = ebosSimulator_.gridView();
+	  auto elemIt = gridView.template begin<0>();
+	  const auto& elemEndIt = gridView.template end<0>();
+	  
+	  //loop over cells in mesh
+	  for (; elemIt != elemEndIt; ++elemIt) {
+	    
+	    const auto& elem = *elemIt;
+	    
+	    //If cell has partition type not equal to Interior change matrix
+	    if (elem.partitionType() != Dune::InteriorEntity)
+	      {
+	
+		//find local id of overlap cell
+		int lcell = lid.id(elem);
+		
+		//loop over block matrix 
+		for (int ii=0;ii<numEq;++ii)
+		  {
+		    for (int jj=0;jj<numEq;++jj)
+		      {
+			//If we are on the diagonal set matrix to "large" value. Otherwise set value to zero
+			if (ii==jj)
+			  {	
+			    ebosJacIgnoreOverlap[lcell][lcell][ii][jj]=diag_val;
+			  }
+			else
+			  {
+			    ebosJacIgnoreOverlap[lcell][lcell][ii][jj]=0.0;
+			  }
+		      }
```

Comment:
This can be replaced with ebosJacIgnoreOverlap[lcell][lcell] = diag_block, see comment above.